scripts/delete-extension.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+source .env
+RELEASE_NAMESPACE="${RELEASE_NAMESPACE:-arc-osm-system}"
+
+export RESOURCEID=subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCEGROUP/providers/Microsoft.Kubernetes/connectedClusters/$CLUSTERNAME
+export HELM_EXPERIMENTAL_OCI=1
+
+jq -n \
+    --arg tag "$CHECKOUT_TAG" \
+    --arg namespace "$RELEASE_NAMESPACE" \
+    '{properties: {extensionType: "Microsoft.openservicemesh", autoUpgradeMinorVersion: "false", version: $tag, releaseTrain: "Staging", scope: { cluster: { releaseNamespace: $namespace } } } }' > osm_extension.json
+
+az account set --subscription=$SUBSCRIPTION
+
+# disable the osm extension
+az rest --method DELETE --uri "https://management.azure.com/$RESOURCEID/providers/Microsoft.KubernetesConfiguration/extensions/osm?api-Version=2020-07-01-preview" --body @osm_extension.json --debug
```

Comment:
you don't need `body` for `DELETE`

NOTE: the extension instance name `osm` preceding `Microsoft.KubernetesConfiguration/extensions/` should be the same as the one you used when running `az rest --method PUT`.

Reply:
I'm confused by the note. I used the same uri in this command as the one used in PUT from add-extension.sh. Am I missing something else? I will remove the `body` in a follow up

Reply:
> I will remove the body in a follow up.
Can you pls remove it in this PR since it's incorrect?

> Are you referring to the osm?api-Version=2020-07-01-preview after Microsoft.KubernetesConfiguration/extensions/? I don't see an extension name before it. I can parameterize this though.

Yes let's make it into a parameter. The string `osm` following `Microsoft.KubernetesConfiguration/extensions/` is the extension instance's name and can be whatever value. This value should match for PUT, DELETE, and should be used to validate the output of GET.

Reply:
> Can you pls remove it in this PR since it's incorrect?
Yes of course, I meant I'd fix it in a follow up commit, not a new PR.

Should `api-Version=2020-07-01-preview` be part of the same parameter, a separate parameter, or will it be constant?

Reply:
I think we should keep the api version constant and use the PR process to update it to ensure api changes are introduced as part of the code.

Reply:
Addressed by https://github.com/Azure/osm-azure/pull/37